pkg/bucketeer/event/processor.go

```diff
@@ -285,25 +285,35 @@ func (p *processor) pushInternalSDKErrorMetricsEvent(ctx context.Context, api mo
 	}
 }
 
-func (p *processor) pushErrorStatusCodeMetricsEvent(ctx context.Context, api model.APIID, code int) {
+func (p *processor) pushErrorStatusCodeMetricsEvent(ctx context.Context, api model.APIID, code int, err error) {
 	var evt interface{}
-	switch code {
-	case http.StatusBadRequest:
+	switch {
+	// Update error metrics report
+	// https://github.com/bucketeer-io/bucketeer/issues/799
+	case 300 <= code && code < 400:
+		evt = model.NewRedirectionRequestErrorMetricsEvent(p.tag, api, code)
+	case code == http.StatusBadRequest:
 		evt = model.NewBadRequestErrorMetricsEvent(p.tag, api)
-	case http.StatusUnauthorized:
+	case code == http.StatusUnauthorized:
 		evt = model.NewUnauthorizedErrorMetricsEvent(p.tag, api)
-	case http.StatusForbidden:
+	case code == http.StatusForbidden:
 		evt = model.NewForbiddenErrorMetricsEvent(p.tag, api)
-	case http.StatusNotFound:
+	case code == http.StatusNotFound:
 		evt = model.NewNotFoundErrorMetricsEvent(p.tag, api)
-	case 499:
+	case code == http.StatusMethodNotAllowed:
+		evt = model.NewInternalSDKErrorMetricsEvent(p.tag, api, err.Error())
+	case code == http.StatusRequestTimeout:
+		evt = model.NewTimeoutErrorMetricsEvent(p.tag, api)
+	case code == http.StatusRequestEntityTooLarge:
+		evt = model.NewPayloadTooLargeErrorMetricsEvent(p.tag, api)
+	case code == 499:
 		evt = model.NewClientClosedRequestErrorMetricsEvent(p.tag, api)
-	case http.StatusInternalServerError:
+	case code == http.StatusInternalServerError:
 		evt = model.NewInternalServerErrorMetricsEvent(p.tag, api)
-	case http.StatusServiceUnavailable:
+	case code == http.StatusServiceUnavailable || code == http.StatusBadGateway:
 		evt = model.NewServiceUnavailableErrorMetricsEvent(p.tag, api)
 	default:
-		evt = model.NewUnknownErrorMetricsEvent(p.tag, api)
+		evt = model.NewUnknownErrorMetricsEvent(p.tag, code, err.Error(), api)
```

Comment:
I Changed to use error as message

Reply:
If it sends status code and an error message, I would not need to create each event because these two information already answer what kind of error we got. WDYT?

Reply:
It's true, but we would need to change the server implementation to support it and keep two implementations for old and new SDKs.
I think it's better to leave it as it is because we don't change the metrics often, but if it is the case we can create an issue to support it in the future.